### PR TITLE
Added streaming endpoint support for streaming client

### PR DESCRIPTION
**Pull request by bstrzele**

### CODE REVIEW

File: demos/common/stream_client/stream_client.py [hunk 1]
```diff
@@ -199,26 +204,36 @@ def start(self, *, ovms_address : str, input_name : str, model_name : str, datat
...
+                    inputs=[grpcclient.InferInput(input_name, np_frame.shape, datatype.string())]
+                    inputs[0].set_data_from_numpy(np_frame)
+                    if streaming_api:
+                        triton_client.async_stream_infer(model_name=model_name, inputs=inputs, parameters={"OVMS_MP_TIMESTAMP":int(cv2.getTickCount() / cv2.getTickFrequency() * 1e6)})
+                    else:
+                        triton_client.async_infer(
+                            model_name=model_name,
+                            callback=partial(self.callback, frame, i, timestamp),
+                            inputs=inputs)
+                    i += 1
```
Comment (atobiszei) [inline demos/common/stream_client/stream_client.py:228]: Rename i -> iteration. In the context of callback method it is not clear what is 'i'. Is it intentional to have first callback with i set to None, and in following ones with properly set?

File: demos/common/stream_client/stream_client.py [hunk 2]
```diff
@@ -199,26 +204,36 @@ def start(self, *, ovms_address : str, input_name : str, model_name : str, datat
...
+        try:
+            while not self.force_exit:
+                timestamp = time.time()
+                frame = self.grab_frame()
+                if frame is not None:
+                    np_frame = np.array([frame], dtype=datatype.dtype()) if batch else np.array(frame, dtype=datatype.dtype())
+                    inputs=[grpcclient.InferInput(input_name, np_frame.shape, datatype.string())]
+                    inputs[0].set_data_from_numpy(np_frame)
+                    if streaming_api:
+                        triton_client.async_stream_infer(model_name=model_name, inputs=inputs, parameters={"OVMS_MP_TIMESTAMP":int(cv2.getTickCount() / cv2.getTickFrequency() * 1e6)})
```
Comment (atobiszei) [inline demos/common/stream_client/stream_client.py:222]:
```suggestion
                        triton_client.async_stream_infer(model_name=model_name, inputs=inputs, parameters={"OVMS_MP_TIMESTAMP":int(cv2.getTickCount() / cv2.getTickFrequency() * 1e6)})
```

extract method calculate timestamp:
int(cv2.getTickCount() / cv2.getTickFrequency() * 1e6)

File: demos/common/stream_client/stream_client.py [hunk 3]
```diff
@@ -199,26 +204,36 @@ def start(self, *, ovms_address : str, input_name : str, model_name : str, datat
             if self.height is None:
                 self.height = np_test_frame.shape[0]
         self.output_backend.init(self.sink, fps, self.width, self.height)
+
+        if streaming_api:
+            triton_client.start_stream(partial(self.callback, None, None, None))
             
         i = 0
```
Comment (atobiszei) [inline demos/common/stream_client/stream_client.py:211]: Shouldn't we use that i=0 in first iteration?

**Review by dtrawins — APPROVED** [review]

File: demos/common/stream_client/stream_client.py [hunk 4]
```diff
@@ -132,18 +143,24 @@ def grab_frame(self):
...
+        self.req_q.get()
 
     def display(self):
         i = 0 
         while True:
-            if self.pq.empty():
-                continue
             entry = self.pq.get()
-            if (entry[0] == i and self.exact) or (entry[0] > i and self.exact is not True):
+            if (entry[0] == i and self.exact and self.streaming_api is not True) or (entry[0] > i and (self.exact is not True or self.streaming_api is True)):
```
Comment (atobiszei) [inline demos/common/stream_client/stream_client.py:163]: Rename i -> displayed_frame_id?
Unpack queue entry:
send_frame_id, received_frame, timestamp = self.pq.get()
Make ifology easier to understand:
(entry[0] == i and self.exact and self.streaming_api is not True)
(entry[0] > i and (self.exact is not True or self.streaming_api is True))

File: demos/common/stream_client/stream_client.py [hunk 5]
```diff
@@ -180,12 +199,15 @@ def start(self, *, ovms_address : str, input_name : str, model_name : str, datat
             Limits how long client could run
         limit_frames : int
             Limits how many frames should be processed
+        streaming_api : bool
+            Use experimental streaming endpoint
         """
 
-        self.cap = cv2.VideoCapture(self.source, cv2.CAP_ANY)
+        self.cap = cv2.VideoCapture(int(self.source) if len(self.source) == 1 and self.source[0].isdigit() else self.source, cv2.CAP_ANY)
```
Comment (dkalinowski) [inline demos/common/stream_client/stream_client.py:206]: I'm not quite sure what is this first digit check logic.
Add a command or document what self.source is

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**